### PR TITLE
Feature parse

### PR DESCRIPTION
**Pull request by dwsideriusNIST**
Proposal for new parser for isotherm_data block:
1. Converts tabs to whitespace (tabs are column delimiters in paste-from-excel, at least on my Mac)
2. Collapses whitespace in case multiple spaces are between columns
3. Uses Pandas to interpret the column data, then sends the output to ndarray

### CODE REVIEW

File: digitizer/parse.py [hunk 1]
```diff
@@ -51,12 +53,16 @@ def prepare_isotherm_dict(form):
...
     data['measurement_type'] = form.inp_measurement_type.value
 
-    measurements = re.sub(' +', ' ', form.inp_isotherm_data.value)
-    measurements = measurements.replace(" ", ",")
-    measurements = np.genfromtxt(StringIO(measurements),
-                                 delimiter=',',
-                                 comments='#')
-    measurements = np.array(measurements, ndmin=2)  # deal with single data row
+    measurements = form.inp_isotherm_data.value  # get string from input text box
+    measurements = measurements.replace('\t', ' ')  # replace tabs with spaces
```
Comment (ltalirz) [inline digitizer/parse.py]: Since you're already replacing tabs with spaces here, can you remove tabs as a delimiter below?

**Review by ltalirz — CHANGES_REQUESTED** [review]
Thanks Dan! 

So, this makes it work for the cases you need?

pandas so far has not been a dependency, i.e. please uncomment it in the setup.py and fix a version.
I've also added the pre-commit check on github actions now; if it's too much bother to adjust the code no problem an I can take care of it as well.

**Comment by ltalirz** [comment]
P.S. Could you ideally provide a couple of examples of text input that it should be able to parse (2-3 should be enough)?
You can simply paste them in comments here. I will then add a test for the parsing function.

File: digitizer/parse.py (hunk 1 above)
Comment (dwsideriusNIST) [reply digitizer/parse.py]: corrected in additional commit

**Comment by ltalirz** [comment]
In case you're wondering about the best course of action here: It should be:

1. Add this repo under a new remote `upstream` (if not done already)
   `git remote add upstream https://github.com/ltalirz/isotherm-digitizer-panel`
2. `git fetch upstream`
3. `git rebase -i upstream/master` (while on your `feature-parse` branch)

This should get you up to date - there may be a minor conflict to resolve in parse.py

**Comment by dwsideriusNIST** [comment]
> P.S. Could you ideally provide a couple of examples of text input that it should be able to parse (2-3 should be enough)?
> You can simply paste them in comments here. I will then add a test for the parsing function.

Not entirely sure this is going work:

Example 1 (paste from MS Excel):

0.03362 | 0.1127
0.08609 | 0.3133
0.15333 | 0.5590
0.46870 | 1.5469
0.72747 | 2.1537
0.92713 | 2.5158
1.07506 | 2.7393
1.21081 | 2.9156
1.40979 | 3.132

That didn't work, as the tab was converted to |. Please try a direct paste from MS Excel if you have it. I've also tested this with LibreOffice Calc, Google Sheets, and Gnumeric, all of which worked fine.

Example 2: using too much white space. (Libreoffice does this.)

0.03362  0.1127   
0.08609   0.3133   
0.15333  0.5590   
0.46870     1.5469   
0.72747   2.1537   

Example 3: 

0.03362 ;  0.1127   
0.08609  ;  0.3133   
0.15333 ;  0.5590   
0.46870  ;   1.5469   
0.72747 ;  2.1537 

Actually, this test fails because the column parser is getting confused by the use of both spaces and semicolons. Hold off on any more testing, as I need to rebase to the master branch and think about additional delimiters.

**Comment by dwsideriusNIST** [comment]
My local branch got mangled by the rebase. I'm going to rewind to the current HEAD of your master, then re-commit the changes. PR may need to be closed.

**Comment by dwsideriusNIST** [comment]
Closing PR because merges got all mixed up during the rebase. New PR will be opened.

**Pull request closed without merging**